system_metrics_collector/src/system_metrics_collector/metrics_message_publisher.hpp

```diff
@@ -49,7 +50,7 @@ metrics_statistics_msgs::msg::MetricsMessage GenerateStatisticMessage(
 /**
  * Simple class to facilitate publishing messages containing statistics data
  */
-class MetricsMessagePublisher
+class MetricsMessagePublisher : virtual public MetricDetailsInterface
```

Comment:
Why is ```virtual``` needed? Here, you shouldn't need to extend ```MetricDetailsInterface``` because Collector->PeriodicMeasurementNode, Collector->TopicStatisticsCollector exposes the metric details.

Reply:
I didn't remove it from here because I'm not sure there won't be a use-case where a class only inherits from `MetricsMessagePublisher` and not `Collector`. In the spirit of [YAGNI](https://martinfowler.com/bliki/Yagni.html) principle, I'm okay removing it.